src/olympia/scanners/tests/test_views.py

```diff
@@ -326,3 +332,29 @@ def test_get_results_with_blocked_versions(self):
 
         response = self.client.get(self.url)
         self.assert_json_results(response, expected_results=5)
+
+    def test_get_results_with_good_blocked_versions(self):
+        self.create_switch('enable-scanner-results-api', active=True)
+        # Result labelled as "good" because auto-approve has been confirmed.
+        version_1 = version_factory(addon=addon_factory())
+        result_1 = ScannerResult.objects.create(
+            scanner=CUSTOMS, version=version_1
+        )
+        VersionLog.objects.create(
+            activity_log=ActivityLog.create(
+                action=amo.LOG.CONFIRM_AUTO_APPROVED,
+                version=version_1,
+                user=self.user,
+            ),
+            version=version_1,
+        )
+        # Oh noes! The version has been blocked.
```

Comment:
Would it be worth adding a check assertion here to verify that at this point the result would be labelled `good`, to make sure that the reason it is labelled `bad` below is valid?

Reply:
It is covered by other test cases so I would say it is fine?

Reply:
Yes, we do generally take that approach, although I must say I'm not a huge fan of it. It does cut down on test code, but I actually think it's better if one can read a single test in isolation and be able to tell that it's correctly testing what needs to be tested. This approach, which, again, I know we commonly take, means that we have to have knowledge of other tests in the file to be sure that our test setup is correct.

Personally, I think one additional check assertion in a test adds more value than it takes away (due to maintenance), but I know this isn't how we usually do things, and this PR isn't necessarily the place to change that approach. In other words, let's leave it as is, but perhaps this is a topic worth discussing further.